lib/puppet/provider/keycloak_realm/kcadm.rb

```diff
@@ -123,11 +143,15 @@ def create
       if self.class.browser_security_headers.include?(property) && !data.key?('browserSecurityHeaders')
         data['browserSecurityHeaders'] = {}
       end
+      if self.class.smtp_server_properties.include?(property)
```

Comment:
This should be `self.class.smtp_server_properties.include?(property) && !data.key?('smtpServer')` and then if this is true it adds the hash `data['smtpServer'] = {}` so that you are only adding the key if SMTP server settings exist. Then this current assignemtn gets moved to line 155.

Reply:
This one has now been done as requested. Did not seem to cause any issues.